### PR TITLE
New eval and seed

### PR DESCRIPTION
**Pull request by discordianfish**

### CODE REVIEW

File: diambra/arena/arena_gym.py [hunk 1]
```diff
@@ -45,89 +47,42 @@ def __init__(self, env_settings):
                                             dtype=np.uint8)
 
     # Processing Environment info
-    def env_info_process(self, env_info):
+    def env_info_process(self, env_info_dict):
```
Comment (discordianfish) [inline diambra/arena/arena_gym.py:50]: Similar q as in engine: Why do we need to translate these and not use the protobuf generated structs directly?

File: diambra/arena/engine/interface.py [hunk 2]
```diff
@@ -17,124 +17,173 @@ def __init__(self, env_address):
         self.channel = grpc.insecure_channel(env_address)
         self.stub = interface_pb2_grpc.EnvServerStub(self.channel)
 
+        # Wait for grpc server to be ready
+        print("Trying to connect to DIAMBRA Engine server ...")
+        try:
+            grpc.channel_ready_future(self.channel).result(timeout=60)
```
Comment (discordianfish) [inline diambra/arena/engine/interface.py:23]: We should make the timeout configurable

File: diambra/arena/engine/interface.py [hunk 3]
```diff
@@ -17,124 +17,173 @@ def __init__(self, env_address):
         self.channel = grpc.insecure_channel(env_address)
         self.stub = interface_pb2_grpc.EnvServerStub(self.channel)
 
+        # Wait for grpc server to be ready
+        print("Trying to connect to DIAMBRA Engine server ...")
```
Comment (discordianfish) [inline diambra/arena/engine/interface.py:21]: Can you use a proper logger for these things, eg https://docs.python.org/3/library/logging.html?

File: diambra/arena/engine/interface.py [hunk 4]
```diff
@@ -17,124 +17,173 @@ def __init__(self, env_address):
...
+        try:
+            grpc.channel_ready_future(self.channel).result(timeout=60)
+        except grpc.FutureTimeoutError:
+            print("... failed.")
+            exceptionMessage = "DIAMBRA Arena failed to connect to the Engine Server."
+            print(exceptionMessage)
```
Comment (discordianfish) [inline diambra/arena/engine/interface.py:27]: No need to print all these things if we raise an exception that prints this error anyway, right?

File: diambra/arena/engine/interface.py [hunk 5]
```diff
@@ -17,124 +17,173 @@ def __init__(self, env_address):
...
-
-        output += end_char
-
-        return output
-
-    # Send env settings, retrieve env info and int variables list
-    def env_init(self, env_settings):
-        env_settings_string = self.env_settings_to_string(env_settings)
+    # Transforming env settings dict to pb request
+    def env_settings_to_pb_request(self, env_settings):
```
Comment (discordianfish) [inline diambra/arena/engine/interface.py:39]: Same q on why we need this translation

File: tests/random_test_gym.py [hunk 6]
```diff
@@ -189,7 +189,7 @@
                                    - n_rounds * (max_continue + 1) * env.max_delta_health,
                                    np.mean(cumulative_ep_rew_all)))
 
-        print("ALL GOOD!")
+        print("COMPLETED SUCCESSFULLY!")
     except Exception as e:
```
Comment (discordianfish) [inline tests/random_test_gym.py:193]: Why do we catch the exception at all? Wouldn't it make sense to just have it terminate the script with non-zero return code instead?

In general we should use a proper testing library but let's address that in some futher iteration.

File: diambra/arena/engine/interface.py (hunk 2 above)
Comment (alexpalms) [reply diambra/arena/engine/interface.py]: should be solved

File: diambra/arena/arena_gym.py (hunk 1 above)
Comment (alexpalms) [reply diambra/arena/arena_gym.py:50]: let's address this in the next PR

File: tests/random_test_gym.py (hunk 6 above)
Comment (alexpalms) [reply tests/random_test_gym.py:193]: Yes we do need a proper testing suite, this is how it works for now, let's keep this in mind for the proper iteration in which we will put in place a complete testing functionality

**Pull request merged**